src/Appwrite/Platform/Tasks/ScheduleExecutions.php

```diff
@@ -38,25 +39,29 @@ protected function enqueueResources(Group $pools, Database $dbForConsole): void
                 continue;
             }
 
-            $now = new \DateTime();
             $scheduledAt = new \DateTime($schedule['schedule']);
-
-            if ($scheduledAt > $now) {
+            if ($scheduledAt <= $intervalEnd) {
                 continue;
             }
 
-            $queueForFunctions
-                ->setType('schedule')
-                // Set functionId instead of function as we don't have $dbForProject
-                // TODO: Refactor to use function instead of functionId
-                ->setFunctionId($schedule['resource']['functionId'])
-                ->setExecution($schedule['resource'])
-                ->setMethod($schedule['data']['method'] ?? 'POST')
-                ->setPath($schedule['data']['path'] ?? '/')
-                ->setHeaders($schedule['data']['headers'] ?? [])
-                ->setBody($schedule['data']['body'] ?? '')
-                ->setProject($schedule['project'])
-                ->trigger();
+            $delay = $scheduledAt->getTimestamp() - (new \DateTime())->getTimestamp();
+
+            \go(function () use ($queueForFunctions, $schedule, $delay) {
+                \sleep($delay);
```

Comment:
Inside non-hooked coroutine it might be [better](https://wiki.swoole.com/en/#/getting_started/notice?id=notes) to use [Co::sleep](https://wiki.swoole.com/en/#/coroutine/system?id=sleep)